packages/query/src/block-processor/index.ts

```diff
@@ -126,20 +122,6 @@ export class BlockProcessor implements BlockProcessorInterface {
     }
   }
 
-  // Compares the locally stored, filtered SCT root with the actual one on chain. They should match.
-  // This is expensive to do every block, so should only be done in development.
-  // @ts-expect-error Only used ad-hoc in dev
-  private async assertRootValid(blockHeight: bigint): Promise<void> {
-    const sourceOfTruth = await this.querier.app.keyValue(`sct/anchor/${blockHeight}`);
-    const inMemoryRoot = this.viewServer.getSctRoot();
-
-    if (!decodeSctRoot(sourceOfTruth).equals(inMemoryRoot)) {
-      throw new Error(
-        `Block height: ${blockHeight}. Wasm root does not match remote source of truth. Programmer error.`,
-      );
-    }
-  }
```

Comment:
It's actually important for us to keep this. In dev mode, we use this on occasion to assert that the filtered state tree we are creating is valid. We do this by querying a root hash and comparing it to our own. Think we need to hunt down where someone moved this query to 🤔

Reply:
the type is now located here

https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.storage.v1alpha1#penumbra.storage.v1alpha1.KeyValueResponse.Value

but i don't think this is accessible via any service we currently implement.

there seem to be some replacement endpoints on the query service

https://buf.build/penumbra-zone/penumbra/docs/main:penumbra.core.app.v1alpha1#penumbra.core.app.v1alpha1.QueryService

Reply:
there are commits today so i think it's in motion

Reply:
Hi @erwanor 👋 , looks like this query was removed: https://github.com/penumbra-zone/penumbra/commit/2b48cf12ff35720c433999f4a0135d18a029fc8b. Is there another place we can make a query to validate our local tree?

Reply:
> Hi @erwanor 👋 , looks like this query was removed: [penumbra-zone/penumbra@2b48cf1](https://github.com/penumbra-zone/penumbra/commit/2b48cf12ff35720c433999f4a0135d18a029fc8b). Is there another place we can make a query to validate our local tree?

The KV RPCs are in the storage query service. I didn't realize this was used here, will flag this next time, sorry about that and thanks for tracking me down!